src/drt/src/pa/FlexPA_init.cpp

```diff
@@ -127,7 +132,13 @@ void FlexPA::initUniqueInstance_main(
   vector<frInst*> ndrInsts;
   vector<frCoord> offset;
   int cnt = 0;
+  std::set<frString> inst_names;
+  for (auto inst : target_insts_)
+    inst_names.insert(inst->getName());
   for (auto& inst : design_->getTopBlock()->getInsts()) {
+    if (!inst_names.empty()
+        && inst_names.find(inst->getName()) == inst_names.end())
```

Comment:
It would be faster to convert the inst_names into frInst* and then just compare pointers here.